Mathlib/Order/Interval/Set/Disjoint.lean

```diff
@@ -65,6 +65,18 @@ theorem Ioc_disjoint_Ioi (h : b ≤ c) : Disjoint (Ioc a b) (Ioi c) :=
 theorem Ioc_disjoint_Ioi_same : Disjoint (Ioc a b) (Ioi b) :=
   Ioc_disjoint_Ioi le_rfl
 
+@[simp] theorem Ioi_disjoint_Iio (h : a ≤ b) : Disjoint (Ioi b) (Iio a) :=
+  disjoint_left.mpr fun _ hx hy ↦ not_lt_of_le h (hx.trans hy)
```

Comment:
This shouldn't be marked `simp`, likewise for the other `a ≤ b` version below. (You could make a `DenselyOrdered` version that is an `iff` if you wanted.)